lib/stages/applyGains.cpp

```diff
@@ -381,6 +323,53 @@ void applyGains::apply_thread() {
 }
 
 
+void applyGains::fetch_thread() {
+
+    auto& dm = datasetManager::instance();
+
+    while (!stop_thread) {
+
+        // Wait until we've start receiving data before fetching any updates. By
+        // waiting until then before reading the gains we can test for
+        // consistency with the incoming data
+        if (!started) {
+            continue;
+        }
+
+        // Block for an update, and check if we should just exit
+        auto update = update_fetch_queue.get();
+        if (!update)
+            break;
+
+        auto [update_id, transition_interval, new_ts] = *update;
+
+        std::optional<applyGains::GainData> gain_data;
+        if (read_from_file) {
+            INFO("Reading gains from file...");
+            gain_data = read_gain_file(update_id);
+        } else {
+            INFO("Fetching gains...");
```

Comment:
If this happens every 10min, that's a lot of not-so-useful output.

Reply:
Errr. I guess the issue is that it might happen every 10s, so it's even worse. That is a good point so I'll turn this down to DEBUG.